src/include/txn_inline.h

```diff
@@ -152,6 +194,27 @@ __txn_resolve_prepared_update(WT_SESSION_IMPL *session, WT_UPDATE *upd)
     WT_PUBLISH(upd->prepare_state, WT_PREPARE_RESOLVED);
 }
 
+/*
+ * __txn_resolve_prepared_page_deleted --
+ *     Resolve a prepared page deleted structure.
+ */
+static inline void
+__txn_resolve_prepared_page_deleted(WT_SESSION_IMPL *session, WT_PAGE_DELETED *page_del)
```

Comment:
Why not a function like: 
```__txn_apply_prepare_state_page_del(WT_SESSION_IMPL *session, WT_PAGE_DELETED *page_del, bool commit) {
     WT_TXN *txn;

     txn = session->txn;

     if (commit) {
        page_del->timestamp  = txn->commit_timestamp;
        page_del->durable_timestamp = txn->durable_timestamp;
        WT_PUBLISH(page_del->prepare_state, WT_PREPARE_RESOLVED)
       } else {
          upd->durable_ts = WT_TS_NONE;
        WT_PUBLISH(upd->prepare_state, WT_PREPARE_INPROGRESS);
       upd->durable_ts = WT_TS_NONE;
        WT_PUBLISH(upd->prepare_state, WT_PREPARE_INPROGRESS);
      }
   }

```

Reply:
I don't want to mix the prepare logic and commit logic in the same function. We do them separately in normal update so I also want to keep them separately here.

Reply:
I think it'll be cleaner in the calling code if we pass commit in, e.g:
```
   for (; *updp != NULL; ++updp)
       txn_apply_prepare_state_update(session, *updp, commit);
  
   txn_apply_prepare_state_page_del(session, ref->page_del, commit);
```

Reply:
I agree with Luke. Change the existing update logic also into a single function. The suggested change is much cleaner than the existing one.

Reply:
done